server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearch.java

```diff
@@ -654,6 +654,25 @@ private void buildCommunicationSummaryStatistics(Map<String, CommunicationSummar
             }
         }
 
+        addNodeInformation(stats, index, criteria, addMetrics, false);
+        addNodeInformation(stats, index, criteria, addMetrics, true);
+    }
+
+    /**
+     * This method adds node information to the communication summary nodes constructed based on the
+     * communication details.
+     *
+     * @param stats The communication summary stats
+     * @param index The index
+     * @param criteria The query criteria
+     * @param addMetrics Whether to add metrics or just discover any missing nodes
+     * @param clients Whether node information should be located for clients (i.e. fragments with
+     *                                  top level Producer node)
```

Comment:
Yes trace fragment. In this particular case, it is looking for a trace fragment that only has a single Producer node - so a special case where the fragment represents a call out to a service.